include/deal.II/numerics/solution_transfer.h

```diff
@@ -373,37 +373,28 @@ class SolutionTransfer
   /**
    * Exception
    */
-  DeclException0(ExcNotPrepared);
+  DeclExceptionMsg(ExcNotPrepared,
+                   "You are attempting an operation for which this object is "
+                   "not prepared. This may be because you either did not call "
+                   "one of the prepare_*() functions at all, or because you "
+                   "called the wrong one for the operation you are currently "
+                   "attempting.");
 
   /**
    * Exception
    */
-  DeclException0(ExcAlreadyPrepForRef);
+  DeclExceptionMsg(ExcAlreadyPrepForRef,
+                   "You are attempting to call one of the prepare_*() functions "
+                   "of this object to prepare it for an operation for which  it "
+                   "it is already prepared.");
 
   /**
    * Exception
    */
-  DeclException0(ExcAlreadyPrepForCoarseAndRef);
-
-  /**
-   * Exception
-   */
-  DeclException0(ExcTriaPrepCoarseningNotCalledBefore);
-
-  /**
-   * Exception
-   */
-  DeclException0(ExcNoInVectorsGiven);
-
-  /**
-   * Exception
-   */
-  DeclException0(ExcVectorsDifferFromInVectors);
-
-  /**
-   * Exception
-   */
-  DeclException0(ExcNumberOfDoFsPerCellHasChanged);
+  DeclExceptionMsg(ExcAlreadyPrepForCoarseAndRef,
+                   "You are attempting to call one of the prepare_*() functions "
+                   "of this object to prepare it for an operation for which  it "
```

Comment:
Use only one space. Maybe remove this exception and replace with ExcAlreadyPrepForRef?